src/browserlib/extract-dfns.mjs

```diff
@@ -225,6 +228,173 @@ export default function (spec, idToHeading = {}) {
     .map(node => definitionMapper(node, idToHeading));
 }
 
+function preProcessEcmascript() {
+  const sectionFilter= ":not([legacy])";
+
+  const wrapWithDfn = (el) => {
+    // wrap with a dfn
+    const dfn = document.createElement("dfn");
+    for (let child of [...el.childNodes]) {
+      dfn.appendChild(child);
+    }
+    el.appendChild(dfn);
+    // set id
+    dfn.setAttribute("id", el.parentNode.getAttribute("id"));
+    dfn.dataset.ltNodefault = true;
+    return dfn;
+  };
+
+  const cleanMethodName = (name) => {
+    return name.replace(/\[/g, '')
+      .replace(/\]/g, '') // removing brackets used to mark optional args
+      .replace(/ \( */, '(')
+      .replace(/ *\)/, ')')
+      .replace(/ *,/g, ','); // trimming internal spaces
+  };
+
+  let definitionNames = new Set();
+
+  const sectionNumberRegExp = /^([A-Z]\.)?[0-9\.]+ /;
+  // any rule needs its exception
```

Comment:
any exception needs an explanation :)

Could you explain in the comment why these exceptions are needed and, perhaps, the conditions that, once met, could remove the need for them?

Reply:
fixed in 44f505c